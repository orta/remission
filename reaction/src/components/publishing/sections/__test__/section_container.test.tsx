import * as React from "react"
import * as renderer from "react-test-renderer"

import "jest-styled-components"

import SectionContainer from "../section_container"
import Text from "../text"

it("renders overflow_fillwidth properly", () => {
  const section = {
    layout: "overflow_fillwidth",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <Text html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})

it("renders fillwidth properly", () => {
  const section = {
    layout: "fillwidth",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <Text html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})

it("renders column_width properly", () => {
  const section = {
    layout: "column_width",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <Text html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})
