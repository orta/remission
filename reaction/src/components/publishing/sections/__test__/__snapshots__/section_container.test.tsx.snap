// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders column_width properly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 680px;
  margin: auto;
  margin-bottom: 40px;
}

.c1 {
  position: relative;
  padding-bottom: ;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #999;
}

.c1 p,
.c1 ul,
.c1 ol {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child {
  padding-top: 0;
}

.c1 p:last-child {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 li {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  -webkit-text-align: center;
  text-align: center;
}

.c1 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h3 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 em {
  font-style: ;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 50px;
  line-height: 1.1em;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
}

.c1 .content-start {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 65px;
  line-height: 1em;
  float: left;
  line-height: .5em;
  margin-right: 10px;
  margin-top: .298em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow:before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

@media (max-width: 1250px) {
  .c0 {
    width: 680px;
  }
}

@media (max-width: 900px) {
  .c0 {
    width: 100%;
    padding: 0px;
    margin: 0 0 40px 0;
  }
}

@media (max-width: 900px) {
  .c1 {
    max-width: calc(100% - 40px);
    margin: 0 auto;
  }
}

@media (max-width: 720px) {
  .c1 {
    max-width: 100%;
  }
}

@media (max-width: 600px) {
  .c1 p,
  .c1 ul,
  .c1 ol {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 40px;
    line-height: 1.1em;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders fillwidth properly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: auto;
  margin-bottom: 40px;
}

.c1 {
  position: relative;
  padding-bottom: ;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #999;
}

.c1 p,
.c1 ul,
.c1 ol {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child {
  padding-top: 0;
}

.c1 p:last-child {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 li {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  -webkit-text-align: center;
  text-align: center;
}

.c1 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h3 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 em {
  font-style: ;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 50px;
  line-height: 1.1em;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
}

.c1 .content-start {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 65px;
  line-height: 1em;
  float: left;
  line-height: .5em;
  margin-right: 10px;
  margin-top: .298em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow:before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

@media (max-width: 1250px) {
  .c0 {
    width: 100%;
  }
}

@media (max-width: 900px) {
  .c0 {
    width: 100%;
    padding: 0px;
    margin: 0 0 40px 0;
  }
}

@media (max-width: 900px) {
  .c1 {
    max-width: calc(100% - 40px);
    margin: 0 auto;
  }
}

@media (max-width: 720px) {
  .c1 {
    max-width: 100%;
  }
}

@media (max-width: 600px) {
  .c1 p,
  .c1 ul,
  .c1 ol {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 40px;
    line-height: 1.1em;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders overflow_fillwidth properly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 780px;
  margin: auto;
  margin-bottom: 40px;
}

.c1 {
  position: relative;
  padding-bottom: ;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #999;
}

.c1 p,
.c1 ul,
.c1 ol {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child {
  padding-top: 0;
}

.c1 p:last-child {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 li {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  -webkit-text-align: center;
  text-align: center;
}

.c1 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h3 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 em {
  font-style: ;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 50px;
  line-height: 1.1em;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
}

.c1 .content-start {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 65px;
  line-height: 1em;
  float: left;
  line-height: .5em;
  margin-right: 10px;
  margin-top: .298em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow:before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

@media (max-width: 1250px) {
  .c0 {
    width: 780px;
  }
}

@media (max-width: 900px) {
  .c0 {
    width: 100%;
    padding: 0px;
    margin: 0 0 40px 0;
  }
}

@media (max-width: 900px) {
  .c1 {
    max-width: calc(100% - 40px);
    margin: 0 auto;
  }
}

@media (max-width: 720px) {
  .c1 {
    max-width: 100%;
  }
}

@media (max-width: 600px) {
  .c1 p,
  .c1 ul,
  .c1 ol {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 40px;
    line-height: 1.1em;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;
